pkg/network/ovn_kubernetes.go

```diff
@@ -532,10 +535,33 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 func shouldRenderIPsec(conf *operv1.OVNKubernetesConfig, bootstrapResult *bootstrap.BootstrapResult) (renderIPsecMachineConfig, renderIPsecDaemonSet,
 	renderIPsecOVN, renderIPsecHostDaemonSet, renderIPsecContainerizedDaemonSet, renderIPsecDaemonSetAsCreateWaitOnly bool) {
 	isHypershiftHostedCluster := bootstrapResult.Infra.HostedControlPlane != nil
-	isIPsecEnabled := conf.IPsecConfig != nil
 	isIpsecUpgrade := bootstrapResult.OVN.IPsecUpdateStatus != nil && bootstrapResult.OVN.IPsecUpdateStatus.LegacyIPsecUpgrade
 	isOVNIPsecActive := bootstrapResult.OVN.IPsecUpdateStatus != nil && bootstrapResult.OVN.IPsecUpdateStatus.OVNIPsecActive
 
+	// Find the IPsec mode from Ipsec.config
+	// Ipsec.config == nil (bw compatibility) || ipsecConfig == Off ==> ipsec is disabled
+	// ipsecConfig.mode == "" (bw compatibility) || ipsec.Config == Full ==> ipsec is enabled for NS and EW
+	// ipsecConfig.mode == External ==> ipsec is enabled for NS only
+
+	mode := operv1.IPsecModeDisabled // Sould stay so if conf.IPsecConfig == nil
+	klog.Infof("IPsec:  Lopoking at ipsecConfig = %+v", conf.IPsecConfig)
+	if conf.IPsecConfig != nil {
+		klog.Infof("IPsec ipsecConfig = %+v", conf.IPsecConfig)
+		if conf.IPsecConfig.Mode != "" {
+			mode = conf.IPsecConfig.Mode
+		} else {
+			mode = operv1.IPsecModeFull // BW compatiniglity with existing configs
+			//TO-REMOVE debug log msg
+			klog.Infof("IPsec mode is not set in ipsecConfig. Assuming upgrade: setting IPsec mode to Full")
+			// For upgrade only - update the object to a valid value
+			conf.IPsecConfig.Mode = operv1.IPsecModeFull
```

Comment:
No, you shouldn't do this. Don't write to the object, leave it as it was until the user decides to make a change. Else you may end up fighting with whatever configuration management system they are using.